docs/source/blocks/block-style-wrapper.md

```diff
@@ -165,6 +170,72 @@ data like:
 
 will generate classnames `primary inverted`
 
+## Inject custom CSS properties
+
+```{versionadded} 17.8.0
+```
+
+The style wrapper also allows to inject custom CSS properties.
+This is useful where the property that you want to inject is customizable per project.
+For example, imagine you have an existing global CSS custom property `--image-aspect-ratio` that you use for all images in all blocks.
+Then in your customized theme, you could set CSS attributes for this property, changing it in runtime.
+The key feature of custom CSS properties is that they can be applied also using the cascade.
+This means that they can be placed anywhere in either CSS definitions or HTML structure, and they will be applied only in the context where they are defined.
+
+As an example, first define the style of a teaser block image as follows.
+
+```css
+.block.teaser img {
+  aspect-ratio: var(--image-aspect-ratio, 16 / 9);
+}
+```
+
+Next, you can enhance a block's schema by injecting the custom CSS property as follows.
+
+```js
+  schema.properties.styles.schema.fieldsets[0].fields = [
+    ...schema.properties.styles.schema.fieldsets[0].fields,
+    '--image-aspect-ratio',
+  ];
+
+  // We use a select widget and set a default
+  schema.properties.styles.schema.properties['--image-aspect-ratio'] = {
+    widget: 'select',
+    title: 'Aspect Ratio',
+    choices: [
+      ['1', '1:1'],
+      ['16 / 9', '16/9'],
+    ],
+    default: '1',
+  };
+```
+
+Finally the markup of the block will contain the custom property as shown.
+
+```html
+<div class="block teaser" style="--image-aspect-ratio: 1">
+...
```

Comment:
Without an `img`, I don't see how the property will be applied because the selector does not actually match anything in the HTML as it is.

```suggestion
<img src="example.png">
```